server/src/test/java/io/spine/core/EventTest.java

```diff
@@ -0,0 +1,219 @@
+import io.spine.base.RejectionMessage;
+import io.spine.base.ThrowableMessage;
+import io.spine.base.Time;
+import io.spine.server.entity.rejection.EntityAlreadyArchived;
+import io.spine.server.entity.rejection.StandardRejections;
+import io.spine.server.event.EventFactory;
+import io.spine.server.type.CommandEnvelope;
+import io.spine.server.type.EventEnvelope;
+import io.spine.server.type.given.GivenEvent;
+import io.spine.test.core.given.EtProjectCreated;
+import io.spine.testing.UtilityClassTest;
+import io.spine.testing.client.TestActorRequestFactory;
+import io.spine.type.TypeName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.spine.protobuf.AnyPacker.pack;
+import static io.spine.server.type.given.EventsTestEnv.commandContext;
+import static io.spine.server.type.given.EventsTestEnv.event;
+import static io.spine.server.type.given.EventsTestEnv.tenantId;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Test functionality of {@link Events} utility class.
+ *
+ * <p>This test suite is placed under the {@code server} module to avoid dependency on the event
+ * generation code which belongs to server-side.
+ */
+@DisplayName("Event should")
+public class EventTest extends UtilityClassTest<Events> {
+
+    private static final TestActorRequestFactory requestFactory =
+            new TestActorRequestFactory(EventTest.class);
+
+    private EventFactory eventFactory;
+
+    private Event event;
+    private EventContext context;
+
+    EventTest() {
+        super(Events.class);
+    }
+
+    @BeforeEach
+    void setUp() {
+        CommandEnvelope cmd = generate();
+        StringValue producerId = StringValue.of(getClass().getSimpleName());
+        eventFactory = EventFactory.on(cmd, Identifier.pack(producerId));
+        event = eventFactory.createEvent(GivenEvent.message(), null);
+        context = event.context();
+    }
+
+    private static CommandEnvelope generate() {
+        return CommandEnvelope.of(requestFactory.generateCommand());
+    }
+
+    @Override
+    protected void configure(NullPointerTester tester) {
+        super.configure(tester);
+        EntityAlreadyArchived defaultThrowableMessage = EntityAlreadyArchived
+                .newBuilder()
+                .setEntityId(Any.getDefaultInstance())
+                .build();
+        tester.setDefault(StringValue.class, StringValue.getDefaultInstance())
+              .setDefault(EventContext.class, GivenEvent.context())
+              .setDefault(Version.class, Version.getDefaultInstance())
+              .setDefault(Event.class, Event.getDefaultInstance())
+              .setDefault(ThrowableMessage.class, defaultThrowableMessage);
+    }
+
+    @Nested
+    @DisplayName("obtain")
```

Comment:
To conform to the English grammar rules, I would modify some of the display names.
Here is would say [Event should] `provide`.